src/main.rs

```diff
@@ -1045,13 +1054,18 @@ fn run() -> Result<(), failure::Error> {
                 let expiration_ttl = put_key_matches
                     .value_of("expiration-ttl")
                     .map(|t| t.to_string());
+                let metadata = put_key_matches
+                    .value_of("metadata")
+                    .map(serde_json::from_str)
+                    .map(Result::unwrap); // success is guaranteed in the argument's `validator` call
```

Comment:
imho the interface for validator is bad (it's throwing away parsed information that you're going to need anyway)... but that shouldn't excuse using bare unwraps

Reply:
I removed the validator and the bare unwrap.